### PR TITLE
Networks added

### PR DESCRIPTION
**Pull request by gguijarro-c-chwy**

### CODE REVIEW

File: lib/core/crypto/dart_esr/src/signing_request_manager.dart [hunk 1]
```diff
@@ -809,31 +809,31 @@ extension HyphaSigningRequestManager on SigningRequestManager {
   /// Returns: Network
   /// throws: unsupported network when the chainID can't be parsed.
   ///
-  static Networks resolveNetwork(List<dynamic> chainId) {
+  static Network resolveNetwork(List<dynamic> chainId) {
```
Comment (gguijarro-c-chwy) [inline lib/core/crypto/dart_esr/src/signing_request_manager.dart:812]: changed name Networks -> Network

File: lib/core/network/api/services/remote_config_service.dart [hunk 2]
```diff
@@ -2,9 +2,25 @@ import 'dart:convert';
 
 import 'package:firebase_remote_config/firebase_remote_config.dart';
 
-enum Networks { telos, telosTestnet, eos, eosTestnet }
+enum Network {
```
Comment (gguijarro-c-chwy) [inline lib/core/network/api/services/remote_config_service.dart:5]: Changed name and added their string value.

Comment (gguijarro-c-chwy) [reply lib/core/network/api/services/remote_config_service.dart:5]: @n13  pls check the enum maps to the correct string value. At the end, when the calls are made, the string is what is being passed to the call

File: lib/core/network/models/user_profile_data.dart [hunk 3]
```diff
@@ -9,6 +10,7 @@ class UserProfileData with _$UserProfileData {
 
   factory UserProfileData({
     required String accountName,
+    required Network network,
```
Comment (gguijarro-c-chwy) [inline lib/core/network/models/user_profile_data.dart:13]: added a network to the UserProfileData

File: lib/core/network/repository/auth_repository.dart [hunk 4]
```diff
@@ -66,7 +67,15 @@ class AuthRepository {
         publicKey: userAuthData.publicKey.toString(),
       );
 
-      _saveUserData(UserProfileData(accountName: accountName, userName: userName), userAuthData, false);
+      _saveUserData(
+        UserProfileData(
+          accountName: accountName,
+          userName: userName,
+          network: Network.fromString(inviteLinkData.chain),
```
Comment (gguijarro-c-chwy) [inline lib/core/network/repository/auth_repository.dart:74]: @n13  This needs your review. inviteLinkData.chain is a String, and i am mapping this to an ENUM
Pls review the enum and the string values . make sure the map will succeed

File: lib/ui/search_user/usecases/search_for_user_use_case.dart [hunk 5]
```diff
@@ -1,22 +1,20 @@
...
-  // FindAvailableAccountUseCase
 
   SearchForMemberUseCase(this.userAccountRepository, this.getUserProfilesFromAccountsUseCase);
 
   Future<Result<List<UserProfileData>, HyphaError>> run(String searchQuery) async {
     final Result<List<String>, HyphaError> result = await userAccountRepository.findHyphaAccounts(searchQuery);
-    if(result.isValue) {
-      return getUserProfilesFromAccountsUseCase.run(result.asValue!.value);
+    if (result.isValue) {
+      return getUserProfilesFromAccountsUseCase.run({Network.eos: result.asValue!.value});
```
Comment (gguijarro-c-chwy) [inline lib/ui/search_user/usecases/search_for_user_use_case.dart:17]: Nik. this is very very important. needs your attention: 

 await userAccountRepository.findHyphaAccounts(searchQuery);

we make a call to get `findHyphaAccounts`

I asumme the findHyphaAccounts are all in the Network.eos ?? and i am setting those accounts to use that network

File: lib/core/network/api/services/remote_config_service.dart [hunk 6]
```diff
@@ -2,9 +2,25 @@ import 'dart:convert';
...
+  telos('telos'),
+  telosTestnet('telosTestnet'),
+  eos('eos'),
+  eosTestnet('eosTestnet');
```
Comment (n13) [inline lib/core/network/api/services/remote_config_service.dart:10]: Why do you need to do that?

If you just left it as it was :
enum Networks { telos, telosTestnet, eos, eosTestnet }

Then you can do Networks.telos.name and that will be == "telos"

https://api.flutter.dev/flutter/dart-core/EnumName/name.html

I'd change this back to what it was.

**Review by n13 — CHANGES_REQUESTED** [review]
Change enum back & use .name property?!

File: lib/core/network/api/services/remote_config_service.dart (hunk 6 above)
Comment (n13) [reply lib/core/network/api/services/remote_config_service.dart:10]: BTW I have used the .name function in the code so I know it works.

File: lib/core/network/api/services/remote_config_service.dart (hunk 2 above)
Comment (n13) [reply lib/core/network/api/services/remote_config_service.dart:5]: Don't need to do that see below

File: lib/core/network/api/services/remote_config_service.dart (hunk 6 above)
Comment (n13) [reply lib/core/network/api/services/remote_config_service.dart:10]: Edit: Also yes I am in favor or renaming it to "Network" without the 's' - that's what it should be. 

Thanks for that ;)

File: lib/core/network/models/user_profile_data.freezed.dart [hunk 7]
```diff
@@ -64,6 +70,10 @@ class _$UserProfileDataCopyWithImpl<$Res, $Val extends UserProfileData>
           ? _value.accountName
           : accountName // ignore: cast_nullable_to_non_nullable
               as String,
+      network: null == network
+          ? _value.network
+          : network // ignore: cast_nullable_to_non_nullable
```
Comment (n13) [inline lib/core/network/models/user_profile_data.freezed.dart:75]: maybe we can default existing profiles to Network.telos - compatibility for old accounts?

File: lib/core/network/repository/auth_repository.dart (hunk 4 above)
Comment (n13) [reply lib/core/network/repository/auth_repository.dart:74]: I see - I guess with the .name we can't do that but still we could make a little mapper that maps these values and defaults to telos if we can't find it.

File: lib/ui/search_user/usecases/search_for_user_use_case.dart (hunk 5 above)
Comment (n13) [reply lib/ui/search_user/usecases/search_for_user_use_case.dart:17]: Actually in this case it should use the network of the user

Ie, if I am logged in / active as "illumination" on Telos, then this find function should find people on the same network, ie, Telos

If I am logged in as "illum1nation" on EOS, then find should find people on EOS ... 

We can't send coins from EOS to Telos & vice versa... at least not easily, although we may add cross chain transactions later, but I think we will add them only for the Hypha token...

File: lib/ui/send/usecases/send_token_use_case.dart [hunk 8]
```diff
@@ -33,7 +33,7 @@ class SendTokenUseCase {
         ),
       ),
       memo: memo ?? '',
-      network: Networks.telos,
+      network: Network.telos,
```
Comment (n13) [inline lib/ui/send/usecases/send_token_use_case.dart:36]: We can fix this later but yeah this should use the user's network for sending (or doing any transactions)

File: lib/core/network/api/aws_amplify/amplify_service.dart [hunk 9]
```diff
@@ -148,7 +148,7 @@ class AmplifyService {
 
       // handle CUSTOM_CHALLENGE challenge
       final loginCode = e.challengeParameters['loginCode'];
-      await eosService.loginWithCode(accountName: accountName, loginCode: loginCode, network: Networks.telos);
+      await eosService.loginWithCode(accountName: accountName, loginCode: loginCode, network: Network.telos);
       print('return challenge $loginCode');
```
Comment (n13) [inline lib/core/network/api/aws_amplify/amplify_service.dart:151]: I guess we can just go over bare usages of constants - telos - and change them all to use the account's network

in a different PR though

**Review by n13 — APPROVED** [review]
LGTM

**Pull request merged**